src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,53 +1,126 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
+import cn from 'classnames';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+// eslint-disable-next-line max-len
+const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+type Props = {
+  onAdd: (post: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [isFormValid, setIsFormValid] = useState(false);
+
+  const reset = () => {
+    setCount(0);
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+    setIsFormValid(false);
+  };
+
+  const handleInputChange = (
+    setter: React.Dispatch<React.SetStateAction<string>>, value: string,
+  ) => {
+    setter(value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const newMovie: Movie = {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    };
+
+    onAdd(newMovie);
+    setCount(count + 1);
+
+    // reset();
+  };
+
+  useEffect(() => {
+    const isImgUrlValid = pattern.test(imgUrl);
+    const isImdbUrlValid = pattern.test(imdbUrl);
+
+    setIsFormValid(!!title && isImgUrlValid && isImdbUrlValid && !!imdbId);
+  }, [title, imgUrl, imdbUrl, imdbId]);
```

Comment:
There is no need in that useEffect. isFormValid can be just a common variable. 
```suggestion
  const isFormValid = !!title && pattern.test(imgUrl) && pattern.test(imdbUrl) && !!imdbId
```

and then in JSX
`<button ... disabled={isFormValid}>`

It should work in the same way